src/NexusMods.App.UI/Pages/NexusModsDataProvider.cs

```diff
@@ -372,6 +374,31 @@ public IObservable<IChangeSet<LoadoutItemModel, EntityId>> ObserveNestedLoadoutI
                     ChildrenObservable = childrenObservable,
                 };
 
+                // Inherit the icon from the first child
+                model.ChildrenObservable.FirstAsync().Subscribe(set =>
+                {
+                    foreach (var item in set)
+                    {
+                        if (item.Reason != ChangeReason.Add) continue;
+                        
+                        // Note(sewer):
+                        // The child may not be activated, so we can't just copy the thumbnail, from it as it may
+                        // not have been loaded yet. We need to manually load it.
+                        var current = item.Current;
+                        var modPageThumbnailPipeline = ImagePipelines.GetModPageThumbnailPipeline(_serviceProvider);
+                        var libraryLinkedItem = LibraryLinkedLoadoutItem.Load(_connection.Db, current.GetLoadoutItemIds().First());
+                        if (libraryLinkedItem.IsValid() && libraryLinkedItem.LibraryItem.TryGetAsNexusModsLibraryItem(out var nexusLibraryItem))
+                        {
+                            ImagePipelines.CreateObservable(nexusLibraryItem.ModPageMetadataId, modPageThumbnailPipeline)
+                                .Take(1)
+                                .ObserveOnUIThreadDispatcher()
+                                .Subscribe(this, (bitmap, _) => model.Thumbnail.Value = bitmap);
+                        }
+                        
+                        return;
+                    }
+                });
+
```

Comment:
This is a complete mess. Just go to `FakeParentLoadoutItemModel` and subscribe to changes there.